discussion_18.md

```diff
@@ -0,0 +1,147 @@
+
+- [x] #9780
+- [ ] #9781
+- [ ] #9782
+- [ ] #9783
+
+<details><summary>
+
+### prior maintainer upgrade guides
+
+</summary>
+
+- #8307
+- #7958
+- #7213
+- #6624
+- #6011
+- #5468
+
+</details>
+
+### Example Diffs
+
+- <https://github.com/dbt-labs/dbt-redshift/compare/v1.7.0...v1.8.0b1>
+- <https://github.com/dbt-labs/dbt-bigquery/compare/v1.7.0...v1.8.0b1>
+- <https://github.com/dbt-labs/dbt-snowflake/compare/v1.7.0...v1.8.0b1>
+- <https://github.com/dbt-labs/dbt-spark/compare/v1.7.0...v1.8.0b1>
+
+## TL;DR
+
+TBD
+
+
+## Decoupled Dependency on dbt-core
+
+### Context
+
+- [dbt Core Roadmap Update 11/23: Adapters & Artifacts](https://github.com/dbt-labs/dbt-core/blob/main/docs/roadmap/2023-11-dbt-tng.md#adapters--artifacts)
+- [dbt-core#9171](https://github.com/dbt-labs/dbt-core/discussions/9171)
+- [dbt-adapters#87](https://github.com/dbt-labs/dbt-adapters/discussions/87)
+
+#### Excerpt from #9171
+
+> The goals of the work are to do the following:
+>
+> - Dramatically lower the effort required to maintain an adapter plugin for dbt-core
+> - Enable users to install the latest dbt-core version, even if their dbt-* adapter has not had a corresponding release
+
+### How to implement
+
+Everything you need to know is in [dbt-adapters#87](https://github.com/dbt-labs/dbt-adapters/discussions/87). If you have a question or concern, please ask it there.
+
+## Unit Testing
+
+Unit testing is a long-awaited feature that is finally coming to dbt. It is a powerful feature that will allow users to test their models in isolation from the rest of their project. This should not require a great deal of work on the part of the adapter maintainer, but, more importantly, how dbt is used changes, so it's important that we have test coverage for these new scenarios.s
+
+### support for `--empty` flag to enable "dry run" mode
+
+- [dbt-core#8980](https://github.com/dbt-labs/dbt-core/issues/8980): original issue
+- [dbt-core#8971](https://github.com/dbt-labs/dbt-core/pull/8971): Core PR
+- [dbt-redshift#666](https://github.com/dbt-labs/dbt-redshift/issues/666): Example implementation
+
+#### How to implement `--empty` support
+
+[dbt-core#8971](https://github.com/dbt-labs/dbt-core/pull/8971) added a new BaseRelation method: [`render_limited()`](https://github.com/dbt-labs/dbt-core/blob/7967be7bb373a3c737196bc0ebbe31ef6f4ed354/core/dbt/adapters/base/relation.py#L198-L205). Effectively, this method will wrap a model's `SELECT` statement into a subquery
+
+```sql
+(select * from {rendered} limit {self.limit}) _dbt_limit_subq
+
+-- if limit is 0 (`where false` for cost saving)
+(select * from {rendered} where false limit 0) _dbt_limit_subq
+```
+
+If your data platform supports `LIMIT` clauses, you have no work to do. However, some SQL dialects (e.g. T-SQL) do not support `LIMIT` clauses. In this case, you will need to implement `render_limited()` for your adapter.
```

Comment:
@amychen1776 FYI dbt-fabric & dbt-synapse will need to do this to support unit testing. AFAICT it isn't currently implemented

Reply:
Thank you for catching this! Would you be able to open up an issue on fabric and synapse adapters to describe what is needed?